### PR TITLE
Use sbt-publish-rsync rather than our own

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: project/Publish.scala [hunk 1]
```diff
@@ -61,27 +63,3 @@ object NoPublish extends AutoPlugin {
...
-
-  object autoImport extends Keys
-  import autoImport._
-
-  override def projectSettings = Seq(
-    deployRsyncArtifacts := List(),
-    deployRsync := {
-      val (_, host) = (Space ~ StringBasic).parsed
-      deployRsyncArtifacts.value.foreach {
-        case (from, to) => s"rsync -rvz $from/ $host:$to" !
```
Comment (patriknw) [inline project/Publish.scala:84]: The new is:
```
Process(Seq("rsync", "-azP", s"$from/", s"$toHost:$toPath"), None, "RSYNC_RSH" -> "ssh -o StrictHostKeyChecking=no")
```

Are the differences ok?

**Review by patriknw — APPROVED** [review]
LGTM, with a sanity check question

File: project/Publish.scala (hunk 1 above)
Comment (raboof) [reply project/Publish.scala:84]: The changes are:

* remove `-r`: already implied by `-a`
* remove `-v`: verbose, seems fine to skip
* add `-a`: seems like an improvement (recursive, keep symlinks, permissions and timestamps)
* add '-P': show progress. not really necessary but OK, could consider dropping upstream
* RSYNC_RSH: not sure, but seems OK
* `StrictHostKeyChecking=no` not sure if needed, could consider dropping upstream

So in short: I don't see anything harmful upstream.

Comment (patriknw) [reply project/Publish.scala:84]: thanks for clarifying

Comment (johanandren) [reply project/Publish.scala:84]: At least on mac `-a` essentially is an alias of `-rlptgoD` - most of those seem ok, but what about `-g` and `-o` which is preserve owner and group. I'm also curious about `-D` which "is the same as --devices and --specials" which I don't find any further details about.

Comment (johanandren) [reply project/Publish.scala:84]: Now I found `--specials`

> This option causes rsync to transfer special files such as named sockets and fifos.

Ehm?

Comment (raboof) [reply project/Publish.scala:84]: I think the group/owner things only apply when you're root, so they don't apply for 'regular use'. I might be wrong, but I didn't notice any problems when using this to publish to gustav

Indeed it wouldn't make sense to rsync over special/device files, but they wouldn't be generated in our docs, so I'm not sure if it matters?

I'd be fine with dropping some of the flags over at sbt-publish-rsync, but perhaps it doesn't help anything?

Comment (johanandren) [reply project/Publish.scala:84]: If it doesn't do anything weird with owner and group I think it's ok

**Review by johanandren — APPROVED** [review]

**Review by ennru — APPROVED** [review]
LGTM.

**Comment by ennru** [comment]
I updated the akka-docs Jenkins build to use `publishRsync` instead of `deployRsync gustav.akka.io` now.

**Comment by ennru** [comment]
That made Jenkins happy.

**Comment by raboof** [comment]
Ah, good one, forgot about that.
